app/models/post.rb

```diff
@@ -18,6 +18,23 @@ class Post < ApplicationRecord
       Arel.sql("posts.likes_count / EXP(DATE_PART('day', NOW() - posts.created_at) / 4.0) DESC")
     )
   }
+  scope :filtered_by_date, lambda { |date_criteria|
+    where('published_at >= ?', date_criteria)
+  }
+  scope :filtered_by_text, lambda { |text|
+    joins(:user).where(
+      Arel.sql('users.first_name LIKE :search OR users.last_name LIKE :search OR users.nickname LIKE :search
```

Comment:
Let's use [CONCAT](https://www.postgresqltutorial.com/postgresql-string-functions/postgresql-concat-function/) for first + last name